### PR TITLE
Add code to detect usb device

### PR DESCRIPTION
**Pull request by nutzhub**
I'm not sure my Kotlin code. which it's converted from Java by tool.
`usbReceiver` is inner-class and it may be better if we can refactor to `receivers` package. But it uses shared var. Please give me a guideline to refactor the code

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -23,9 +23,11 @@
             android:theme="@style/FullscreenTheme">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN"/>
-
+                <action android:name="android.hardware.usb.action.USB_DEVICE_ATTACHED" />
```
Comment (smashdown) [inline app/src/main/AndroidManifest.xml:26]: you might missed 'feature' of USB hosting.
<uses-feature android:name="android.hardware.usb.host" />

see the document below
https://developer.android.com/guide/topics/connectivity/usb/host.html#manifest

File: app/src/main/java/com/nut/kiosk/ui/MainActivity.kt [hunk 2]
```diff
@@ -68,9 +72,48 @@ class MainActivity : AppCompatActivity(), View.OnTouchListener {
...
 
+    private lateinit var usbPermissionIntent: PendingIntent
+    private lateinit var usbManager: UsbManager
+
+    private val usbReceiver = object : BroadcastReceiver() {
+        override fun onReceive(context: Context, intent: Intent) {
+            if (ACTION_USB_PERMISSION == intent.action) {
+                synchronized(this) {
+                    val usbDevice = intent.getParcelableExtra<UsbDevice>(UsbManager.EXTRA_DEVICE)
+                    usbManager.requestPermission(usbDevice, usbPermissionIntent)
```
Comment (smashdown) [inline app/src/main/java/com/nut/kiosk/ui/MainActivity.kt]: you sending permission request on receiver.
but receiver is not working until you got a permission.

so, you should request permission first.
and then onReceive() method you could do something you want.

Comment (smashdown) [reply app/src/main/java/com/nut/kiosk/ui/MainActivity.kt]: and there is no permission callback when user allow it or deny it.
refer to the way to implement it below.
https://developer.android.com/training/permissions/requesting.html

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (nutzhub) [reply app/src/main/AndroidManifest.xml:26]: 👍

**Pull request merged**